base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/dialogs/MultipleRecentConfigsSelectionDialog.java

```diff
@@ -0,0 +1,120 @@
+
+/*
+* This file is part of the ISIS IBEX application.
+* Copyright (C) 2012-2017 Science & Technology Facilities Council.
+* All rights reserved.
+*
+* This program is distributed in the hope that it will be useful.
+* This program and the accompanying materials are made available under the
+* terms of the Eclipse Public License v1.0 which accompanies this distribution.
+* EXCEPT AS EXPRESSLY SET FORTH IN THE ECLIPSE PUBLIC LICENSE V1.0, THE PROGRAM 
+* AND ACCOMPANYING MATERIALS ARE PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES 
+* OR CONDITIONS OF ANY KIND.  See the Eclipse Public License v1.0 for more details.
+*
+* You should have received a copy of the Eclipse Public License v1.0
+* along with this program; if not, you can obtain a copy from
+* https://www.eclipse.org/org/documents/epl-v10.php or 
+* http://opensource.org/licenses/eclipse-1.0.php
+*/
+
+package uk.ac.stfc.isis.ibex.ui.configserver.dialogs;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Shell;
+
+import uk.ac.stfc.isis.ibex.configserver.configuration.ConfigInfo;
+import uk.ac.stfc.isis.ibex.ui.dialogs.SelectionDialog;
+
+/**
+ * Dialog for asking the user to select a multiple configurations or components.
+ */
+public class MultipleRecentConfigsSelectionDialog extends SelectionDialog {
+    
+    /**
+     * The collection of the available configurations/components for the user to
+     * select from.
+     */
+    protected final Collection<ConfigInfo> available;
+
+    /**
+     * Is the dialog to do with components? (as opposed to configs)
+     */
+    protected boolean isComponent;
+
+    /**
+     * The currently selected items.
+     */
+    protected Collection<String> selected = new ArrayList<>();
+    
+    /**
+     * Include the current config in the list of available items.
+     */
+    protected boolean includeCurrent;
+
+    /**
```

Comment:
This comment suggests that `extraListOptions` should be a list of some sort and not an integer. Is this the number of additional options to be applied?